data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/S3SinkConfigTest.java

```diff
@@ -0,0 +1,130 @@
+    private ObjectOptions objectOptions;
+
+    public S3SinkConfigTest() {
+        s3SinkConfig = new S3SinkConfig();
+        thresholdOptions = new ThresholdOptions();
+        objectOptions = new ObjectOptions();
+    }
+    
+    @BeforeEach
+    void setUp() {
+        
+        s3SinkConfig = mock(S3SinkConfig.class);
+        thresholdOptions = mock(ThresholdOptions.class);
+        
+        when(s3SinkConfig.getMaxConnectionRetries()).thenReturn(5);
+        when(s3SinkConfig.getMaxUploadRetries()).thenReturn(5);
+        
+        when(s3SinkConfig.getThresholdOptions()).thenReturn(thresholdOptions);
+        
+        when(thresholdOptions.getEventCollectionDuration()).thenReturn(Duration.ofSeconds(DURATION));
+        
+        when(thresholdOptions.getMaximumSize()).thenReturn(ByteCount.parse(DEFAULT_BYTE_CAPACITY));
+    }
+
+    @Test
+    void max_connection_retries_test() throws NoSuchFieldException, IllegalAccessException {
+        reflectivelySetField(s3SinkConfig, "maxConnectionRetries", MAX_CONNECTION_RETRIES);
+        assertThat(s3SinkConfig.getMaxConnectionRetries(), equalTo(MAX_CONNECTION_RETRIES));
+    }
+
+    @Test
+    void max_upload_retries_test() throws NoSuchFieldException, IllegalAccessException {
+        reflectivelySetField(s3SinkConfig, "maxUploadRetries", MAX_UPLOAD_RETRIES);
+        assertThat(s3SinkConfig.getMaxUploadRetries(), equalTo(MAX_UPLOAD_RETRIES));
+    }
+
+    @Test
+    void default_key_pattern_test() {
+        assertThat(objectOptions.getFilePattern(), equalTo(DEFAULT_KEY_PATTERN));
+    }
+
+    @Test
+    void event_count_duration_test() {
+        assertThat(thresholdOptions.getEventCollectionDuration(),
+                equalTo(Duration.ofSeconds(DURATION)));
+    }
+
+    @Test
+    void default_byte_capacity_test() {
+        assertThat(thresholdOptions.getMaximumSize().getBytes(),
+                equalTo(ByteCount.parse(DEFAULT_BYTE_CAPACITY).getBytes()));
+    }
+
+    @Test
+    void default_request_timeout_test() {
+        assertThat(thresholdOptions.getEventCount(), equalTo(DEFAULT_EVENT_COUNT));
+    }
+
+    @Test
+    void get_bucket_name_test() {
+        assertNull(s3SinkConfig.getBucketName());
+    }
+
+    @Test
+    void get_temporary_storage() {
+        assertThat(s3SinkConfig.getTemporaryStorage(), equalTo(null));
+    }
+
+    @Test
+    void get_threshold_options_in_sinkconfig_exception() {
+        assertNotNull(s3SinkConfig.getThresholdOptions());
+    }
+
+    @Test
+    void get_AWS_Auth_options_in_sinkconfig_exception() {
+        assertNull(s3SinkConfig.getAwsAuthenticationOptions());
+    }
+
+    @Test
+    void default_get_key_path_prefix_test() {
+        assertNull(s3SinkConfig.getKeyPathPrefix());
+    }
+
+    @Test
+    void default_get_json_codec_test() {
+        assertNull(s3SinkConfig.getCodec());
+    }
+
+    private void reflectivelySetField(final S3SinkConfig s3SinkConfig, final String fieldName, final Object value)

```

Comment:
Data Prepper has a utility for this. Please use this instead.

https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-test-common/src/main/java/org/opensearch/dataprepper/test/helper/ReflectivelySetField.java

Reply:
As per updated code, the above ReflectivelySetField is not required.